File: examples/javascript/create_poll.js
const axios = require('axios');

const ENDPOINT = 'https://api.strawpoll.com/v3'
const API_KEY = 'YOUR_API_KEY'

let poll = {
	"title": "Is this a good JavaScript example?",
	"media": null,
	"poll_options": [
		{
			"type": "text",
			"value": "Yes"
		},
		{
			"type": "text",
			"value": "No"
		},
		{
			"type": "text",
			"value": "I don't know"
		}
	],
	"poll_config": {
		"vote_type": "default",
		"allow_comments": true,
		"allow_indeterminate": false,
		"allow_other_option": false,
		"custom_design_colors": null,
		"deadline_at": null,
		"duplication_checking": "ip",
		"edit_vote_permissions": "nobody",
		"force_appearance": null,
		"hide_participants": false,
		"is_multiple_choice": true,
		"multiple_choice_min": 1,
		"multiple_choice_max": 3,
		"number_of_winners": 1,
		"randomize_options": false,
		"require_voter_names": false,
		"results_visibility": "always",
		"use_custom_design": false
	},
	"poll_meta": {
		"description": "I do write some JavaScript myself.",
		"location": null,
	},
	"type": "multiple_choice",
}

axios.post(ENDPOINT + '/polls', 
	poll,
	{
		headers:
			{
				'Content-Type': 'application/json',
				'X-API-KEY': API_KEY 
			},
	}
)
.then(res => {
	console.log(res.data.url) // res.data is a Poll object
})
.catch(error => {
	console.error(error)
})

File: examples/python/create_poll.py
import requests

ENDPOINT = "https://api.strawpoll.com/v3"
API_KEY = "YOUR_API_KEY"

payload = {
	"title": "Is this a good Python example?",
	"media": None,
	"poll_options": [
		{
			"type": "text",
			"value": "Yes"
		},
		{
			"type": "text",
			"value": "No"
		},
		{
			"type": "text",
			"value": "I don't know"
		}
	],
	"poll_config": {
		"is_private": False,
		"vote_type": "default",
		"allow_comments": True,
		"allow_indeterminate": False,
		"allow_other_option": False,
		"custom_design_colors": None,
		"deadline_at": None,
		"duplication_checking": "ip",
		"allow_vpn_users": False,
		"edit_vote_permissions": "nobody",
		"force_appearance": None,
		"hide_participants": False,
		"is_multiple_choice": False,
		"multiple_choice_min": None,
		"multiple_choice_max": None,
		"number_of_winners": 1,
		"randomize_options": False,
		"require_voter_names": False,
		"results_visibility": "always",
		"use_custom_design": False
	},
	"poll_meta": {
		"description": "I don't write much Python myself.",
		"location": None,
	},
	"type": "multiple_choice",
}

response = requests.post(ENDPOINT + '/polls', json = payload, headers = { 'X-API-KEY': API_KEY })

if response:
	poll = response.json() # response is Poll object
	print(poll["url"])
else:
	error = response.json()
	print(error)
